core/src/test/scala/weaponregex/parser/ParserJSTest.scala

```diff
@@ -18,14 +18,14 @@ class ParserJSTest extends ParserTest {
     val parsedTree = Parser(pattern, parserFlavor).get
 
     assert(clue(parsedTree).isInstanceOf[Concat])
-    parsedTree.children foreach (child => assert(!clue(child).isInstanceOf[Boundary]))
+    parsedTree.asInstanceOf[Node].children foreach (child => assert(!clue(child).isInstanceOf[Boundary]))
```

Comment:
Move `asInstanceOf[Node]` to `val` definition for consistency with others 😉 
But there's a chance that a casting exception may occur before any assertion if the parser produces a `Leaf`.